packages/atlas/src/components/_crt/MarketDrawer/steps/MarketStep.tsx

```diff
@@ -0,0 +1,193 @@
+import { FC, useCallback, useEffect } from 'react'
+import { Controller, useForm } from 'react-hook-form'
+
+import { SvgActionPlay } from '@/assets/icons'
+import { ActionDialogButtonProps } from '@/components/ActionBar'
+import { FlexBox } from '@/components/FlexBox'
+import { NumberFormat } from '@/components/NumberFormat'
+import { ColumnBox } from '@/components/ProgressWidget/ProgressWidget.styles'
+import { Text } from '@/components/Text'
+import { TextButton } from '@/components/_buttons/Button'
+import { CrtMarketForm } from '@/components/_crt/MarketDrawer/MarketDrawer.types'
+import { Checkbox } from '@/components/_inputs/Checkbox'
+import { FormField } from '@/components/_inputs/FormField'
+import { TextArea } from '@/components/_inputs/TextArea'
+import { TokenInput } from '@/components/_inputs/TokenInput'
+import { atlasConfig } from '@/config'
+import { useConfirmationModal } from '@/providers/confirmationModal'
+import { useJoystream } from '@/providers/joystream'
+
+type MarketStepProps = {
+  setPrimaryButtonProps: (props: ActionDialogButtonProps) => void
+  setSecondaryButtonProps: (props: ActionDialogButtonProps) => void
+  tokenName: string
+  onClose: () => void
+  onNextStep: (props: CrtMarketForm) => void
+}
+
+const DEFAULT_MIN_PRICE = 100
+
+export const MarketStep: FC<MarketStepProps> = ({
+  tokenName,
+  setPrimaryButtonProps,
+  onNextStep,
+  setSecondaryButtonProps,
+  onClose,
+}) => {
+  const { tokenPrice } = useJoystream()
+  const {
+    control,
+    handleSubmit,
+    resetField,
+    watch,
+    formState: { isDirty, errors },
+  } = useForm<CrtMarketForm>({
+    defaultValues: {
```

Comment:
This is kinda a problem since useForm is mounted directly on the step and step is getting unmounted if you proceed, data is lost when going back from summary to form. 

You would need to pass submitted data to the form, so you can define default values correctly and use predefined price and tnc as a fallback.

Reply:
@WRadoslaw good catch, thank you!

Reply:
@WRadoslaw good catch, thank you!

Reply:
@WRadoslaw good catch, thank you!